addon/route-handler.js

```diff
@@ -41,12 +42,26 @@ function createHandler({ verb, schema, serializerOrRegistry, rawHandler, options
   return handler;
 }
 
+function getModelName(fullPath) {
+  let path = fullPath.split('/');
+  let lastPath;
+  while (path.length > 0) {
+    lastPath = path.splice(-1)[0];
+    if (lastPath && lastPath !== ':id') {
+      break;
+    }
+  }
+  let modelName = dasherize(camelize(singularize(lastPath)));
+  return modelName;
+}
+
 export default class RouteHandler {
 
-  constructor({ schema, verb, rawHandler, customizedCode, options, serializerOrRegistry }) {
+  constructor({ schema, verb, rawHandler, customizedCode, options, path, serializerOrRegistry }) {
     this.verb = verb;
     this.customizedCode = customizedCode;
     this.serializerOrRegistry = serializerOrRegistry;
+    rawHandler = rawHandler || getModelName(path);
```

Comment:
The problem with putting this here is that `getModelName` is a special "ability" of the shorthand route handlers. So let's keep it within the shorthands. It shouldn't be here, because function and object handlers have no notion of a modelName for a path.

So instead, let's pass along `path` through `createHandler`, give it to the shorthand route handlers, and then they can do their special magic probably in base shorthand. It will still be at construction time (you could even set it as state on the route handler, `this.modelClass = this.getModelClassFromPath(path)`.

Also I prefer these functions to be methods on the class, I find it easier to notice when a new object needs to be made if there are a lot of related methods on the class. The free-standing functions makes that a bit harder, I think.